beluga/include/beluga/sensor/data/occupancy_storage_mixin.hpp

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_SENSOR_DATA_OCCUPANCY_STORAGE_MIXIN_HPP
+#define BELUGA_SENSOR_DATA_OCCUPANCY_STORAGE_MIXIN_HPP
+
+#include <cstdint>
+
+#include <ciabatta/ciabatta.hpp>
+#include <sophus/se2.hpp>
+
+/*
+ * \file
+ * \brief Concepts and abstract implementations of occupancy grids.
+ */
+
+namespace beluga {
+
+/// Occupancy grid storage interface layer.
+/**
+ * \tparam Mixin Mixin type
+ * \tparam ValueType Occupancy grid cell value type
+ * \tparam ValueTraitsType Occupancy grid cell value traits type
+ * \tparam MapStorageType Occupancy grid storage type
+ */
+template <class Mixin, typename ValueType, typename ValueTraitsType, typename MapStorageType>
+class OccupancyStorageMixin : public Mixin {
```

Comment:
@glpuga meta: what makes it an occupancy grid? I think we are lacking type contracts in documentation for value related types.